src/System.Runtime.Environment/src/OSName.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.Runtime.InteropServices
+{
+    public struct OSName : IEquatable<OSName>
+    {
+        private readonly string _name;
+
+        private const string WindowsName = "WINDOWS";
+        private const string LinuxName = "LINUX";
+        private const string OSXName = "OSX";
+
+        public OSName(string name)
+        {
+            _name = name;
```

Comment:
I would validate the name argument here. At a minimum check for null (`ArgumentNullException`) and also probably empty (`ArgumentException`).